client/src/partials/exchange/modal.js

```diff
@@ -3,19 +3,34 @@ angular.module('bhima.controllers')
 
 ExchangeModalController.$inject = [
   'ExchangeRateService', 'CurrencyService',
-  'SessionService', '$uibModalInstance', 'data'
+  'SessionService', '$uibModalInstance', 'data', 'exchange'
```

Comment:
You should use `ExchangeRateService` instead.  `exchange` is deprecated, and will hopefully be removed after this PR lands in master.